src/features/HelpAndKnowledge/SolutionsHelp/SolutionDetails/Solutions/Generic/_components/ContractorForm/index.tsx

```diff
@@ -56,24 +65,45 @@ const ContractorForm = ({
   const {
     control,
     handleSubmit,
-    reset,
     formState: { isSubmitting, isDirty, isValid }
   } = methods;
 
-  const { selectedSolution } = useModalSolutionState();
-  const { showMessage } = useMessage();
-  const [create] = useCreateMtoCommonSolutionContractorMutation();
-  const [update] = useUpdateMtoCommonSolutionContractorMutation();
+  const [create] = useCreateMtoCommonSolutionContractorMutation({
+    refetchQueries: [
+      {
+        query: GetMTOSolutionContacts
+      }
+    ]
+  });
+
+  const [update] = useUpdateMtoCommonSolutionContractorMutation({
+    refetchQueries: [
+      {
+        query: GetMTOSolutionContacts
+      }
+    ]
+  });
+
   const [mutationError, setMutationError] = useState<
     'duplicate' | 'generic' | null
   >(null);
+
   const disabledSubmitBtn = isSubmitting || !isDirty || !isValid;
 
+  useEffect(() => {
+    setDisableButton(disabledSubmitBtn);
+  }, [setDisableButton, disabledSubmitBtn]);
+
+  useEffect(() => {
+    setSubmitForm(() => onSubmit);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
We can avoid this needing to override the hook linter.  We need to add in the correct dependencies then wrap our onSubmit in a useCallback

```
useEffect(() => {
    setDisableButton(disabledSubmitBtn);
  }, [setDisableButton, disabledSubmitBtn]);

  const onSubmit = useCallback(
    (formData: TeamMemberFormValues) => {
      if (!selectedSolution) {
        return;
      }

```

You will also need to move the early return and hook underneath the onSubmit

```
  useEffect(() => {
    setSubmitForm(() => onSubmit);
  }, [setSubmitForm, onSubmit]);

  if (!selectedSolution) {
    return null;
  }
```